[ActionSheet] Add properties to allow for theming

codeman7 opened the pull request:
This PR adds properties to the components to allow for theming.

Closes #4903

codeman7 committed:
Intial commit

codeman7 committed:
Intial commit

codeman7 committed:
Update image

codeman7 committed:
Update image

randallli committed:
Automatic changelog preparation for release.

codeman7 committed:
Add cell color

codeman7 committed:
Add cell color

codeman7 committed:
Revert example

codeman7 committed:
Revert example

material-automation commented:
## API diff detected the following changes

### ActionSheet

#### MDCActionSheetController

*new* property: `actionColor` in `MDCActionSheetController`

*new* property: `imageColor` in `MDCActionSheetController`

*new* property: `primaryTextColor` in `MDCActionSheetController`

*new* property: `secondaryTextColor` in `MDCActionSheetController`

*modified* property: `backgroundColor` in `MDCActionSheetController`

| Type of change: | Declaration |
|---|---|
| From: | `@property(nonatomic, nonnull, strong) UIColor *backgroundColor` |
| To: | `@property (readwrite, strong, nonatomic, nonnull) UIColor *backgroundColor;` |

<!-- identifier: apidiff -->

codeman7 commented on components/ActionSheet/src/MDCActionSheetController.m:
```diff
@@ -261,6 +262,13 @@ - (UITableViewCell *)tableView:(UITableView *)tableView
   cell.actionFont = self.actionFont;
+  cell.actionColor = self.actionColor;
+  for (UIView *view in cell.contentView.subviews) {
+    if ([view isKindOfClass:[UIImageView class]]) {
```
This isn't optimal but in order to set the tint color you have to wait until the view has loaded. If you do this in `layoutSubviews` of `MDCActionSheetItemTableViewCell` then it doesn't get set by default. The only other solution I could think of was a flag `imageColorHasBeenSet` in `MDCActionSheetItemTableViewCell` if that was `NO` then set the color to black.

jverkoey replied:
Can we set the tintColor on the cell's contentView? `tintColor` is a recursive API so any subviews should adopt it automatically.

codeman7 replied:
That worked.

yarneo committed:
updated to newest CbC standard (#4956)

Move to the new CbC standard released in CatalogByConvention v2.5.0. See PR: material-foundation/cocoapods-catalog-by-convention#27 for more info on the change.

randallli committed:
[Snackbar] Remove a test that fails on internal infrastructure.

randallli committed:
Hand-modified CHANGELOG.md API diff.

randallli committed:
Bumped version number to 61.0.0.

randallli committed:
Revert "[Snackbar] Remove a test that fails on internal infrastructure."

This reverts commit 5078cf787bc6163ef7ce2b0ae7b97055f37e48b3.

randallli committed:
hand modified changelog again

randallli committed:
Merge branch 'release-candidate' into develop

codeman7 committed:
[ActionSheet] Always show first option (#4963)

This makes it so that if there are too many actions to show on screen when the ActionSheet is presented then always show the first one. Previously the first couple actions would be hidden behind the header.

Closes #4948 

| Before  |   After  |
| ------- | ------- |
|![simulator screen shot - iphone x - 2018-08-30 at 10 24 51](https://user-images.githubusercontent.com/7131294/44858531-36ad1f80-ac40-11e8-93bb-abcdb7107f0a.png)| ![simulator screen shot - iphone x - 2018-08-30 at 10 25 19](https://user-images.githubusercontent.com/7131294/44858564-4c224980-ac40-11e8-8c56-b98c2f73a86f.png)|

codeman7 committed:
[ActionSheet] Add accessibility identifier (#4944)

Add accessibility identifier to `MDCActionSheetAction`.

jverkoey committed:
[scripts/release] Allow umbrellas and files to generate info from a given sha. (#4961)

The umbrellas command is fed in to the apidiff tool as a filter for which components to look at.

Prior to this change, the umbrellas command was ignoring the apidiff base sha and always generating results from origin/stable.

After this change, the umbrellas and files commands will both accept an optional base sha from which to generate results.

Tested by running:

```bash
./scripts/release umbrellas HEAD
# Should output nothing
```

jverkoey committed:
[kokoro] Extract xcode-select logic to a method. (#4962)

We'll be reusing this logic for other jobs in the future.

I also simplified the logic to remove duplication-checking because we just care to select the first version of Xcode that matches our desired version.

fluxminds committed:
[Tabs] fix TabBarViewControllerExample to respect safe area (#4949)

closes #3728 

### Before

![b1](https://user-images.githubusercontent.com/8836258/44799553-351d2200-ab82-11e8-9548-f0eb54cf2769.png)

### After:

![a1](https://user-images.githubusercontent.com/8836258/44799555-364e4f00-ab82-11e8-8aec-6b6316374f9b.png)

codeman7 committed:
[ActionSheet] Update test to guard against silent fail (#4969)

Previously if there was no table the test would pass this addition now guards that if there is no table then the test should fail.

jverkoey committed:
[scripts/release] Add a dragon to publishing the podspec. (#4971)

fluxminds committed:
[Catalog] Avoid cropping on the ShadowLayer example (#4884)

cropped on small device; fix the topLayout issue (the top label covered by the app bar)

closes (https://github.com/material-components/material-components-ios/issues/386)


Before the change:
![simulator screen shot - iphone 5 - 2018-08-24 at 10 18 58](https://user-images.githubusercontent.com/8836258/44589752-59928c00-a787-11e8-8f21-1f56b180c3d5.png)
![simulator screen shot - iphone 5 - 2018-08-24 at 10 18 53](https://user-images.githubusercontent.com/8836258/44589747-57c8c880-a787-11e8-91ea-7b529c6c2835.png)



After the change:
![simulator screen shot - iphone 5 - 2018-08-24 at 10 20 18](https://user-images.githubusercontent.com/8836258/44589762-61523080-a787-11e8-99b4-c39828d52d07.png)
![simulator screen shot - iphone 5 - 2018-08-24 at 10 20 22](https://user-images.githubusercontent.com/8836258/44589765-62835d80-a787-11e8-8ce6-2316176b71a5.png)

jverkoey committed:
Revert "[FlexibleHeader] Add new canAlwaysExpandToMaximumHeight behavior. (#4794)" (#4976)

This reverts commit 2b3722f7b8cc7df131a8b33695990c99931c0e1b.

codeman7 committed:
[ActionSheet] Add typography themer (#4966)

Add a typography themer to the Action Sheet component.

Commit:
[BottomNavigation] Update ripple color for unselected items. (#4950)

All items should share the same (ink) ripple color, based on the
`selectedItemTintColor`. For currently-unselected items, the ripple color
would only be updated if the `unselectedItemTintColor` property was set.

The problem was caused by earlier behavior where the ripple color would be
derived from the current state of the button. If an unselected item were
pressed, its ripple color would be based on the `unselectedItemTintColor`.
That behavior was changed so that the `selectedItemTintColor` was used for the
ripple color in all states, but the logic was not moved outside of the
state-based conditional logic.

||Animated Screenshot|
|--|--|
|Before | ![bn-ink-before](https://user-images.githubusercontent.com/1753199/44802070-bd9ec100-ab88-11e8-86c9-48658f8adcc0.gif)|
|After | ![bn-ink-after](https://user-images.githubusercontent.com/1753199/44802075-c2fc0b80-ab88-11e8-872f-ad586408b928.gif)|

Closes #4937

jverkoey committed:
[scripts/github-comment] Remove duplicate target. (#4975)

This resolves a warning that was appearing when building/running the github-comment project:

```
warning: Ignoring duplicate product 'github-comment'
```

Tested by running:

```
cd scripts/github-comment/
swift run github-comment
```

And verifying that the above warning did not appear and that the command successfully built and executed.

jverkoey commented on components/ActionSheet/src/MDCActionSheetController.h:
```diff
@@ -138,8 +138,31 @@ __attribute__((objc_subclassing_restricted))
+/**
+ The color applied to the message of Alert Controller.
+ */
+@property(nonatomic, strong, nullable) UIColor *messageColor;
```
As discussed in https://github.com/material-components/material-components-ios/pull/4968#discussion_r214355546, I'm a bit worried that this API may not let us build a themer that guarantees a correct theme in cases where properties are set after theming. Is there another way we can structure this API to better support the themer?

codeman7 replied:
Think I found a solution.

jverkoey commented on components/ActionSheet/src/private/MDCActionSheetItemTableViewCell.m:
```diff
@@ -105,6 +105,7 @@ - (void)commonMDCActionSheetItemViewInit {
   [self.contentView addSubview:_imageView];
   _imageView.translatesAutoresizingMaskIntoConstraints = NO;
   _imageView.alpha = ImageAlpha;
+  [_imageView setTintColor:[UIColor blackColor]];
```
If you remove this and rely on recursive tintColor being applied will it still work?

codeman7 replied:
Done

jverkoey committed:
[FlexibleHeader] Add buttons to the configurator demo for shifting the header on/off-screen. (#4979)

Screenshot:

![simulator screen shot - iphone x - 2018-08-31 at 09 13 25](https://user-images.githubusercontent.com/45670/44914406-25c2e380-acfe-11e8-8c79-8d583b5b75a6.png)

codeman7 committed:
Update API

codeman7 committed:
Update implementation

codeman7 committed:
Fix conflicts
